gpflow/config/__config__.py

```diff
@@ -41,11 +49,59 @@ def name(self):
         return f"GPFLOW_{super().name}"
 
 
-def default(value: _Values):
+def _default(value: _Values):
     """Checks if value is set in the environment."""
     return os.getenv(value.name, default=value.value)
 
 
+def _default_numeric_type_factory(valid_types, enum_key, type_name):
+    value = _default(enum_key)
+    if value in valid_types.values():
+        return value
+    if value not in valid_types:
+        raise TypeError(f"Config cannot recognize {type_name} type.")
+    return valid_types[value]
+
+
+def _default_int_factory():
+    valid_types = dict(int16=np.int16, int32=np.int32, int64=np.int64)
+    return _default_numeric_type_factory(valid_types, _Values.INT, "int")
+
+
+def _default_float_factory():
+    valid_types = dict(float16=np.float16, float32=np.float32, float64=np.float64)
+    return _default_numeric_type_factory(valid_types, _Values.FLOAT, "float")
+
+
+def _default_jitter_factory():
+    try:
+        value = float(_default(_Values.JITTER))
+    except ValueError:
+        raise TypeError("Config cannot set the jitter value with non float type.")
+    return value
+
+
+def _default_positive_bijector_factory():
+    bijector_type = _default(_Values.POSITIVE_BIJECTOR)
+    if bijector_type not in positive_bijector_type_map().keys():
+        raise TypeError("Config cannot set the passed value as a default positive bijector.")
+    return bijector_type
+
+
+def _default_positive_minimum_factory():
+    try:
+        value = _default(_Values.POSITIVE_MINIMUM)
+        if value is not None:
+            value = float(value)
```

Comment:
What happens with `None`? If that's a useful, intended case: shouldn't the empty string (`export GPFLOW_POSITIVE_MINIMUM=`) also map to `None`?

Reply:
I think this should never happen, "GPFLOW_POSITIVE_MINIMUM=" must be a number, otherwise don't set it. There is also some interference with other values and how they behave. The empty string might mean different things depending on the context. E.g. None or just default value. I think the rule of thumb should be: _set it or get the hardcoded default value_.

Reply:
but the hardcoded default value is also `None`?...

Reply:
`None` is a special case, exists only for switching off `shift` bijector. Maybe we need to be even more restrictive and set default value to 0 instead of `None`. That is a slightly different question.